### ISSUE TITLE
ESlint rule to prevent proxy desctructuring

### ISSUE THREAD

**Issue by Siegrift** [TSOptics/optics#5]
As mentioned in [docs](https://www.optics.page/docs/Introduction/core-concepts/), destructuring the optic will not work.

I believe this could this be solved by creating an ESlint rule preventing this.

**Comment by VincentQuillien** [comment]
It's a good idea !

I've never written an ESlint rule though, is it possible to statically determine the type of a variable (with a TS integration maybe) ?

**Comment by Siegrift** [comment]
Yeah, I had something like that in mind. It would work only with TS, but that's OK. That said, I haven't written ESlint rule myself either :).

**Comment by 123joshuawu** [comment]
I think its possible to get the type with the `typescript-eslint` parser: https://typescript-eslint.io/developers/custom-rules#typed-rules

Might give this a shot when I have some free time
